src/webgpu/shader/validation/parse/loop.spec.ts

```diff
@@ -0,0 +1,59 @@
+export const description = `Validation parser tests for 'loop' statements`;
+
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { keysOf } from '../../../../common/util/data_tables.js';
+import { ShaderValidationTest } from '../shader_validation_test.js';
+
+export const g = makeTestGroup(ShaderValidationTest);
+
+const kTests = {
+  break: { wgsl: `loop { break; }`, pass: true },
+  return: { wgsl: `loop { return; }`, pass: true },
+  break_continuing: { wgsl: `loop { break; continuing {} }`, pass: true },
+  var_break: { wgsl: `loop { var a = 1; break; }`, pass: true },
+  var_break_continuing_inc: {
+    wgsl: `loop { var a = 1; break; continuing { a += 1; }}`,
+    pass: true,
+  },
+  var_break_continuing_discard: {
+    wgsl: `loop { var a = 1; break; continuing { discard; }}`,
+    pass: true,
+  },
+
+  continue: { wgsl: `loop { continue; }`, pass: false },
+  discard: { wgsl: `loop { discard; }`, pass: false },
+  expr_break: { wgsl: `loop expr { break; }`, pass: false },
+  loop: { wgsl: `loop`, pass: false },
+  empty: { wgsl: `loop{}`, pass: false },
+  continuing_break: { wgsl: `loop { continuing {} break; }`, pass: false },
+  break_continuing_continue: { wgsl: `loop { break; continuing { continue; }`, pass: false },
+  break_continuing_return: { wgsl: `loop { break; continuing { return; }`, pass: false },
+  break_continuing_if_break: {
+    wgsl: `loop { break; continuing { if true { break; } }`,
+    pass: false,
+  },
+  break_continuing_if_return: {
+    wgsl: `loop { break; continuing { if true { return; } }`,
+    pass: false,
+  },
```

Comment:
I assume more extensive 'break if' tests are coming, as part of their own test file.

Reply:
They already exist: https://github.com/gpuweb/cts/blob/main/src/webgpu/shader/validation/parse/break_if.spec.ts